test/agentchat/contrib/vectordb/test_mongodb.py

```diff
@@ -0,0 +1,135 @@
+    - Case 2. the collection exists, if overwrite is True, it will overwrite the collection.
+    - Case 3. the collection exists and overwrite is False, if get_or_create is True, it will get the collection, otherwise it raise a ValueError.
+    """
+    # test_create_collection: case 1
+    if collection_name not in db.list_collections():
+        collection = db.create_collection(
+            collection_name=collection_name,
+            index_name="my_index",
+            similarity="cosine",
+            overwrite=False,
+            get_or_create=True,
+        )
+        assert collection.name == collection_name
+    # test_create_collection: case 2
+    # test overwrite=True
+    collection = db.create_collection(
+        collection_name=collection_name,
+        index_name="my_index_1",
+        similarity="cosine",
+        overwrite=True,
+        get_or_create=True,
+    )
+    assert collection.name == collection_name
+
+    # test_create_collection: case 3
+    # test overwrite=False
+    # test get_or_create=False
+    with pytest.raises(ValueError):
+        collection = db.create_collection(
+            collection_name, index_name="my_index_1", similarity="cosine", overwrite=False, get_or_create=False
+        )
+    # test get_or_create=True
+    collection = db.create_collection(
+        collection_name, index_name="my_index_1", similarity="cosine", overwrite=False, get_or_create=True
+    )
+    assert collection.name == collection_name
+
+    # test_get_collection
+    collection = db.get_collection(collection_name)
+    assert collection.name == collection_name
+
+    # test_insert_docs
+    docs = [{"content": "doc1", "id": "1"}, {"content": "doc2", "id": "2"}, {"content": "doc3", "id": "3"}]
+    db.insert_docs(docs, collection_name, upsert=False)
+    res = list(db.get_collection(collection_name).find({"id": {"$in": ["1", "2"]}}))
+    final_results = [result.get("content") for result in res]
+    assert final_results == ["doc1", "doc2"]
+
+    # test_update_docs
+    docs = [{"content": "doc11", "id": "1"}, {"content": "doc2", "id": "2"}, {"content": "doc3", "id": "3"}]
+    db.update_docs(docs, collection_name)
+    res = list(db.get_collection(collection_name).find({"id": {"$in": ["1", "2"]}}))
+    final_results = [result.get("content") for result in res]
+    assert final_results == ["doc11", "doc2"]
+
+    # test_delete_docs
+    ids = ["1"]
+    db.delete_docs(ids, collection_name)
+    res = list(db.get_collection(collection_name).find({"id": {"$in": ids}}))
+    final_results = [result.get("content") for result in res]
+    assert final_results == []
+
+    # sleep for a few seconds -- make sure vector search index is ready
+    time.sleep(30)
+    # test_retrieve_docs
+    """
+    [[({'content': 'doc2', 'id': '2'}, 1.0),
+    ({'content': 'doc3', 'id': '3'}, 0.9205596446990967)],
+    [({'content': 'doc3', 'id': '3'}, 1.0),
+    ({'content': 'doc2', 'id': '2'}, 0.9205596446990967)]]
```

Comment:
This is similarity instead of distance.

Reply:
Got it! 

<img width="756" alt="Screenshot 2024-06-21 at 10 04 47 AM" src="https://github.com/microsoft/autogen/assets/6230773/861d93f0-0240-49c2-8102-3d5759eaff7b">